src/awesome/class.rs

```diff
@@ -0,0 +1,163 @@
+//! Utility methods and constructors for Lua classes
+
+use std::default::Default;
+use std::rc::Rc;
+use rlua::{self, Lua, ToLua, Table, UserData, AnyUserData, Value, Function};
+use super::object::Object;
+use super::property::Property;
+
+pub type Allocator = Rc<Fn(&Lua) -> rlua::Result<Object>>;
+pub type Collector = Rc<Fn(Object)>;
+pub type PropF<'lua> = rlua::Function<'lua>;
```

Comment:
Why not Rc?

Reply:
Because it's only used in one place, in the properties. Which is just a temporary type that isn't stored in lua directly except as a table which is what the properties struct deserializes to (with `ToLua`).

Moved to properties though.